packages/labs/context/src/test/context-provided_test.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {LitElement} from 'lit';
+import {property} from 'lit/decorators/property.js';
+
+import {ContextProvider, createContext} from '../lit-context';
+import {assert} from '@esm-bundle/chai';
+import {contextProvided} from '../lib/decorators/context-provided';
+
+const simpleContext = createContext<number>('simple-context');
+
+class SimpleContextProvider extends LitElement {
+  private provider = new ContextProvider(this, simpleContext, 1000);
+
+  public setValue(value: number) {
+    this.provider.setValue(value);
+  }
+}
+
+class SimpleContextConsumer extends LitElement {
+  @contextProvided({context: simpleContext})
+  @property({type: Number})
+  public onceValue = 0;
+
+  @contextProvided({context: simpleContext, multiple: true})
+  @property({type: Number})
+  public multipleValue = 0;
+}
+
+customElements.define('simple-context-consumer', SimpleContextConsumer);
+customElements.define('simple-context-provider', SimpleContextProvider);
+
+suite('context-provider', () => {
+  let provider: SimpleContextProvider;
+  let consumer: SimpleContextConsumer;
+
+  setup(async () => {
+    const container = document.createElement('div');
+    container.innerHTML = `
+       <simple-context-provider>
+         <simple-context-consumer></simple-context-consumer>
+       </simple-context-provider>
+     `;
```

Comment:
What if we want many `<simple-context-consumer></simple-context-consumer>` with some way of discriminating each of them?

```html
<simple-context-provider>
    <simple-context-consumer data-discriminant="foo"></simple-context-consumer>
    <simple-context-consumer data-discriminant="bar"></simple-context-consumer>
    <simple-context-consumer data-discriminant="baz"></simple-context-consumer>
</simple-context-provider>
```

That pattern would be useful in a micro-front-end app that knows a list of ids, but can't know data to fill each `<simple-context-consumer />`.

That's what I've been trying to do that prompted me to talk about it in https://github.com/webcomponents-cg/community-protocols/issues/21#issuecomment-934603937

Reply:
Here is an example benjamind/lit#1 for this use-case

Reply:
I think we've agreed that this sort of interaction can be built atop the basic protocol based on what is being passed here. So resolving this conversation for now.